### PR TITLE
HPCC-21391 ESP logging user's password

### PR DESCRIPTION
**Pull request by RussWhitehead**
In CSoapService::processRequest(), the code queries a pointer to the context
userID .  In the call to processHeader(), that ctx pointer gets changed to point
to the password (or possibly something else). If a SOAPCALL error occurs, the logging
attempts to log the userID, but it now points to the pwd. This PR copies the user
ID to a local and references that instead

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by RussWhitehead** [comment]
@wangkx Please review

File: esp/bindings/SOAP/Platform/soapservice.cpp [hunk 1]
```diff
@@ -185,7 +185,8 @@ int CSoapService::processRequest(ISoapMessage &req, ISoapMessage& resp)
     EndTag etag;
     StringBuffer peerStr;
     ctx->getPeer(peerStr);
-    const char* userId = ctx->queryUserId();
+    StringBuffer uid(ctx->queryUserId());
+    const char* userId = uid.str();
```
Comment (wangkx) [inline esp/bindings/SOAP/Platform/soapservice.cpp]: @RussWhitehead looks like the userID is changed after the processHeader() call. You may need to retrieve the userID after that call to make sure that userID is correct.

**Comment by RussWhitehead** [comment]
@wangkx Please rereview

**Comment by wangkx** [comment]
@RussWhitehead looks fine.

**Comment by RussWhitehead** [comment]
@richardkchapman Please merge into 7.0.12

**Pull request merged**